### DISCUSSION TITLE
Is there a config to create separate PR per package?

### DISCUSSION THREAD

**Discussion by shan-du** [renovatebot/renovate#24071] [Request Help] [answered]
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us what version of Renovate you run.

36.57

### If you're self-hosting Renovate, select which platform you are using.

gitlab.com

### Was this something which used to work for you, and then stopped?

I never saw this working

### Wanted end result.

Is there any config settings that can tell renovate-bot to generate separate PRs per package?

For example, if i have a `package.json` file with these dependencies:
```javascript
  "dependencies": {
    "axios": "^0.26.0",
    "react": "^17.0.2",
    "react-dom": "^17.0.2",
    "react-router-dom": "^5.1.2"
  },
```

Is there a way to configure renovate to automatically generate a PR for EACH package upgrade? something like:
```
MR1: Upgrade axios v0.26.0 => v0.26.1
MR2: Upgrade react v17.0.2 => v17.1.0
MR3: etc.
```

I know I can use `packageRules` to list out all the packages with hard-coded names, but that's obviously not ideal & not scalable

### What you tried so far.

I tried to Google for this topic but there's no useful results, neither does this discussion forum

And as mentioned above, I did use `packageRules` config with hard coded names, like this:
```javascript
"packageRules": [
    {
      "matchPackageNames": ["react"],
      "rangeStrategy": "bump"
    },
    {
      "matchPackageNames": ["axios"],
      "rangeStrategy": "bump"
    },
    ...
  ],
```

but this is somehow quite buggy & doesn't work all the time, more over, it's not dynamic & not scalable

### Relevant debug logs

<details><summary>Logs</summary>

```
Copy/paste the relevant log(s) here, between the starting and ending backticks
```

</details>

**Comment by lmilbaum** [comment]
This one worked for my go major deps. Pay attention to the groupName attribute.
```
    {
      "description": "Group go major upgrades",
      "matchManagers": ["gomod"],
      "matchUpdateTypes": ["major"],
      "groupName": null
    }
```

**Reply by shan-du** [reply]
thank you for the suggestion, assuming this goes into `packageRules`, then `"groupName": null` didn't seem to do the trick for me unfortunately :(

**Reply by viceice** [reply]
please provide the config

**Reply by shan-du** [reply]
Hi, I think [rarkins](https://github.com/rarkins) provided a solution which SHOULD work, just gotta fix something on our end to fully test it out.

Thank you guys for the awesome support!

**Reply by pfrayer** [reply]
This solution worked for me, thanks :+1:

**Comment by rarkins** [comment]
Renovate's default behavior is to have one PR per package. If you're seeing different, it's probably because of the `config:recommended` preset and automatic grouping of react packages via `group:recommended`: https://docs.renovatebot.com/presets-config/#configrecommended

If you have one PR per package then you're likely to find a lot of broken PRs though as many packages need to be updated at the same time as their siblings

**Reply by shan-du** [reply]
thank you for the quick reply, we don't have any of those presets though... this is what our `renovate.json` looks like
```
{
  "$schema": "https://docs.renovatebot.com/renovate-schema.json",
  "extends": ["config:base"],
  "baseBranches": ["main"],
  "bumpVersion": "minor",
  "packageRules": [
    {
      "excludePackagePatterns": ["^package-name1", "^@package-name2"],
      "rangeStrategy": "bump"
    }
  ],
  "reviewers": ["@username1", "@username2"],
  "reviewersSampleSize": 1,
  "prBodyTemplate": "{{{header}}}{{{table}}}{{{warnings}}}{{{notes}}}{{{changelogs}}}{{{footer}}}",
  "osvVulnerabilityAlerts": true,
  "transitiveRemediation": true
}

```

**Reply by rarkins** [reply, marked as answer]
`config:base` is the old name for `config:recommended`.

You can try adding `"ignorePresets": ["group:monorepos", "group:recommended"]` to your config to disable all grouping as a result of `config:recommended`.

**Reply by shan-du** [reply]
I think this is working! I'm seeing individual PRs per package coming in!

But there might be something off on the PR generation process coz it never generates more than 2 PRs, but this is likely some problems on our side.

Thank you for all the suggestions tho!

**Reply by galusben** [reply]
Thanks for this thread, I had the same issue and got it resolved by @lmilbaum 's answer.
But I am now hitting the issue where I have only 2 PRs at a time.
Did you get this resolved?

**Reply by galusben** [reply]
Resolved the issue of only 2 PRs by following config:
```
 prConcurrentLimit: 20,
 branchConcurrentLimit: 20,
 prHourlyLimit: 20,
```
